src/components/elements/AddUser/AddUser.tsx

```diff
@@ -10,119 +20,336 @@ interface FuncProps {
 
 const AddUser: FunctionComponent<FuncProps> = (props) => {
 
+    //Store Data
+    const adminDepartments = useSelector((state: RootStateOrAny) => state.AllAdminDepatments);
+    const allAdminRoles = useSelector((state: RootStateOrAny) => state.AllAdminRoles);
+
+
     const [validated, setValidated] = useState(false)
+    const [state, setState] = useState({
+        cwid: "",
+        email: "",
+        firstName: "",
+        lastName: "",
+        middleName: "",
+        department: "",
+        division: "",
+        title: "",
+        description: ""
+    })
+    const { cwid, email, firstName, lastName, middleName, division, title } = state;
+    const [selectedRoles, SetSelectedRoles] = useState([]);
+    const [selectedDepartments, SetSelectedDepartments] = useState([]);
+    const [loading, setLoading] = useState(false);
 
-    const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
-        const form = event.currentTarget;
-        if (form.checkValidity() === false) {
-            event.preventDefault();
-            event.stopPropagation();
-        }
+    const router = useRouter()
+    const isEdit = router.query.userId ? true : false;
+
+    const dispatch = useDispatch();
+
+    const handleValueChangeTargetValue = (inputName, e) => {
+        setState(state => ({ ...state, [inputName]: e.target.value }))
+    }
 
+    const handleSubmit = async (event) => {
+        event.preventDefault();
+        const form = event.currentTarget;
+        // if (form.checkValidity() === false) {
+        //     event.preventDefault();
+        //     event.stopPropagation();
+        // }
         setValidated(true);
+
+        if (form && form.checkValidity() === true) {
```

Comment:
does this checkValidity() function check if any of the fields are empty? Because the fields by default have value '' that might not be considered as empty

Reply:
Fixed. Please check new commit.